### PR TITLE
[trie] Add sortedlist

### PR DESCRIPTION
**Pull request by Liuhaai**

### CODE REVIEW

File: db/trie/mptrie/branchnode.go [hunk 1]
```diff
@@ -35,9 +34,10 @@ func newBranchNode(
 			dirty: true,
 		},
 		children: children,
+		indices:  NewSortList(children),
 	}
-	bnode.cacheNode.serializable = bnode
```
Comment (CoderZhi) [inline db/trie/mptrie/branchnode.go:39]: revert

File: db/trie/mptrie/branchnode.go [hunk 2]
```diff
@@ -52,10 +52,10 @@ func newEmptyRootBranchNode(mpt *merklePatriciaTrie) *branchNode {
 			mpt: mpt,
 		},
 		children: make(map[byte]node),
+		indices:  NewSortList(nil),
 		isRoot:   true,
 	}
 	bnode.cacheNode.serializable = bnode
-
```
Comment (CoderZhi) [inline db/trie/mptrie/branchnode.go:58]: revert

File: db/trie/mptrie/branchnode.go [hunk 3]
```diff
@@ -203,16 +200,13 @@ func (b *branchNode) child(key byte) (node, error) {
 	if !ok {
 		return nil, trie.ErrNotExist
 	}
-
```
Comment (CoderZhi) [inline db/trie/mptrie/branchnode.go:206]: revert

**Review by CoderZhi — APPROVED** [review]

File: db/trie/mptrie/sortedlist.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+func (sl *sortedList) Insert(key uint8) {
+	i := sort.Search(len(sl.li), func(i int) bool {
+		return sl.li[i] >= uint8(key)
+	})
+	if i == len(sl.li) {
+		sl.li = append(sl.li, key)
+	} else {
+		if sl.li[i] != key {
```
Comment (dustinxie) [inline db/trie/mptrie/sortedlist.go:36]: this should always be true? b/c the condition to call `Insert` is only key does not exist in children now
```
if _, exist := b.children[key]; !exist {
			b.indices.Insert(key)
		}
```

Comment (dustinxie) [reply db/trie/mptrie/sortedlist.go:36]: also `make lint`, it may ask you to add comments for these funcs

**Review by dustinxie — APPROVED** [review]

**Pull request merged**